### DISCUSSION TITLE
How to create provider package?

### DISCUSSION THREAD

**Discussion by makrushin-evgenii** [apache/airflow#16688] [Q&A]
Following the [docs](https://airflow.apache.org/docs/apache-airflow-providers/#how-to-create-your-own-provider), I tried to create my  own provider package: https://github.com/makrushin-evgenii/apache-airflow-providers-kontur

I am going to use it to configure Airflows like this:
```
AIRFLOW__CORE__XCOM_BACKEND: 'apache_airflow_providers_kontur.xcoms.drive2.Drive2XCom'
```

Now I am faced with the problem of installing the module and I do not understand what was wrong:
```
airflow-init_1       | [2021-06-28 10:45:12,882] {configuration.py:455} ERROR - No module named 'apache_airflow_providers_kontur'
a
```

Can you help with this? An example of custom provider packages would help a lot

**Comment by potiuk** [comment]
You  have wrong package name. The "apache-airflow-providers-NNNNN" is the name used by the "airflow-community-managed" providers (and in fact the package is "apache.airflow.providers.NNNN" and files are placed in `apache/airflow/providers/NNNN` folder.

In your case, you can choose whatever package you want. In your case you have "apache-airflow-providers-kontur" package it seems (but then you refer to it as "apache_airflow_providers_kontur" (with underscores).

I'd suggest you choose another package name (like "kontur.airflow.provider" for example) and then you should place your files in `kontur/airflow/providers` folder (and place appropriate `__init__.py` files in those folders. 

This is just a regular python package, nothing special with it.
